core/src/components/datetime/datetime.tsx

```diff
@@ -2139,12 +2139,16 @@ export class Datetime implements ComponentInterface {
                 }}
               >
                 <div
+                  part={`calendar-day${isActive ? ' active' : ''}${isToday ? ' today' : ''}`}
                   class="calendar-day-highlight"
                   style={{
                     backgroundColor: dateStyle?.backgroundColor,
+                    color: dateStyle?.textColor,
                   }}
-                ></div>
-                {text}
+                >
+                  {text}
+                </div>
+                <span aria-hidden="true" style={{ opacity: '0' }}>{text}</span>
```

Comment:
hacky, but this ensures the grid rows keep the same height as they have in `main`